### PR TITLE
Leverage UBSan instead of ftrapv for integer overflow checking

### PR DESCRIPTION
**Pull request by shaobo-he**
This PR address the issue #428.

@zvonimir I wanted to add some regressions but wasn't quite sure where to put them.

### CODE REVIEW

File: lib/smack/IntegerOverflowChecker.cpp [hunk 1]
```diff
@@ -124,7 +124,8 @@ bool IntegerOverflowChecker::runOnModule(Module& m) {
         if (auto chkshft = dyn_cast<CallInst>(&*I)) {
 	  Function* chkfn = chkshft->getCalledFunction();
           if (chkfn && chkfn->hasName() &&
-              chkfn->getName().find("__ubsan_handle_shift_out_of_bounds") != std::string::npos) {
+              (chkfn->getName().find("__ubsan_handle_shift_out_of_bounds") != std::string::npos ||
+               chkfn->getName().find("__ubsan_handle_divrem_overflow") != std::string::npos)) {
             // If the call to __ubsan_handle_shift_out_of_bounds is reachable,
```
Comment (keram88) [inline lib/smack/IntegerOverflowChecker.cpp]: In the should just say these functions are only called by ubsan when an error occurs.

**Comment by zvonimir** [comment]
@shaobo-he : could you please run SVCOMP overflow benchmarks to make sure this is ok?

**Comment by zvonimir** [comment]
@shaobo-he : gentle reminder - please run SVCOMP overflow benchmarks on this branch.

**Comment by shaobo-he** [comment]
> @shaobo-he : gentle reminder - please run SVCOMP overflow benchmarks on this branch.

I ran these benchmarks and the results look OK to me. So I think we can merge it.

**Comment by zvonimir** [comment]
Please squash and write a nice commit message. Use `Closes` to refer to the issue it is addressing.

**Comment by shaobo-he** [comment]
> Please squash and write a nice commit message. Use `Closes` to refer to the issue it is addressing.

Done.

**Pull request merged**